src/scene/materials/standard-material.js

```diff
@@ -768,19 +788,25 @@ function _defineTex2D(name, uv, channels, defChannel, vertexColor, detailMode) {
     // construct the transform uniform
     const mapTiling = `${name}MapTiling`;
     const mapOffset = `${name}MapOffset`;
+    const mapRotation = `${name}MapRotation`;
     const mapTransform = `${name}MapTransform`;
     defineUniform(mapTransform, (material, device, scene) => {
         const tiling = material[mapTiling];
         const offset = material[mapOffset];
-        if (tiling.x === 1 && tiling.y === 1 && offset.x === 0 && offset.y === 0) {
+        const rotation = material[mapRotation];
+
+        if (tiling.x === 1 && tiling.y === 1 &&
+            offset.x === 0 && offset.y === 0 &&
+            rotation === 0) {
             return null;
         }
 
-        const uniform = material._allocUniform(mapTransform, () => new Float32Array(4));
-        uniform[0] = tiling.x;
-        uniform[1] = tiling.y;
-        uniform[2] = offset.x;
-        uniform[3] = 1.0 - tiling.y - offset.y;
+        const uniform = material._allocUniform(mapTransform, () => new Float32Array(9));
+        uniform.set([
+            Math.cos(rotation) * tiling.x, Math.sin(rotation) * tiling.x, 0,
+            -Math.sin(rotation) * tiling.y, Math.cos(rotation) * tiling.y, 0,
+            offset.x, 1.0 - tiling.y - offset.y, 1
```

Comment:
I'm not sure this math is correct. You need to build scale and rotation matrix and multiply them together, which is different than multiply only matching elements of the mates. But I agree it'd be good to write optimized version. Perhaps we need Mat3.setTRS, similarly to Mat4.setTRS

Reply:
The math is correct.